### PR TITLE
Update Developer Guide with new Logic, UI, Model and Storage implementations

### LINKED ISSUE #327: Update Model and UI diagrams and implementation in the Developer guide

### PR DESCRIPTION
**Pull request by delishad21**
Currently, our user guide still shows the old implementation details for UI and Model. Lets update it so that future developers can easily understand how to build upon our application.

Fixes #327

### CODE REVIEW

**Review by Zer0Legion — COMMENTED** [review]
Great addition. Many aspects of the new extension were added.

**Review by jayllo-c — COMMENTED** [review]
the implementation explanations are highly detailed and well elaborated

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -75,12 +75,25 @@ The UI consists of a `MainWindow` that is made up of parts e.g.`CommandBox`, `Re
 
 The `UI` component uses the JavaFx UI framework. The layout of these UI parts are defined in matching `.fxml` files that are in the `src/main/resources/view` folder. For example, the layout of the [`MainWindow`](https://github.com/se-edu/addressbook-level3/tree/master/src/main/java/seedu/address/ui/MainWindow.java) is specified in [`MainWindow.fxml`](https://github.com/se-edu/addressbook-level3/tree/master/src/main/resources/view/MainWindow.fxml)
 
-The `UI` component,
+The `CommandBox` takes in user input which is passed onto the `Logic` component for the user input to be parsed and executed. A `CommandResult` is returned after execution and the feedback is displayed to the user through the `ResultDisplay` component of the UI.
+
+For the updating of other components in the UI, after each command execution, `MainWindow` runs an update that calls the update method on `PersonListPanel`, `ExamListPanel` and `StatusBarFooter`.
+
+`PersonListPanel` and `ExamListPanel` update themselves by retriving the `filteredPersonList` and `examList` from the `Model` component and updating the displayed lists accordingly.
```
Comment (danielleloh) [inline docs/DeveloperGuide.md:82]: retriving should be retrieving

Comment (Zer0Legion) [reply docs/DeveloperGuide.md:82]: Wonderful catch

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -103,18 +116,34 @@ Here are the other classes in `Logic` (omitted from the class diagram above) tha
...
 <puml src="diagrams/DeleteSequenceDiagram.puml" alt="Interactions Inside the Logic Component for the `delete 1` Command" />
 
 **Note:** The lifeline for `DeleteCommandParser` should end at the destroy marker (X) but due to a limitation of PlantUML, the lifeline continues till the end of diagram.
 </box>
 
-The sequence diagram below illustrates a more in-depth view of the interactions within the 'Logic' component.
-It takes a user input into the UI, `add n|Dohn Joe p|98765432 a|123 e|dohn@gm.com m|A1234567X s|S1 r|R1`, as an example.
+<div id="specificParsing"></div>
+
+The following is a more detailed explaination on how user input is parsed into a `Command` object (Not mentioned above for simplicity).
```
Comment (danielleloh) [inline docs/DeveloperGuide.md:134]: good detailed explanation

**Pull request merged**